chat-plugins/github.js

```diff
@@ -0,0 +1,93 @@
+/*
+	Github plugin by Spandamn
+	This is based on https://github.com/Ecuacion/Pokemon-Showdown-Node-Bot/blob/master/features/github/index.js
+	Yes, some of the code is copied so credits to xfix, Ecuacion, jd, and anyone who's name I forgot to put here
+*/
+'use strict';
+
+let config = {};
+
+if (!Config.github) {
+	return;
+} else if (Config.github === {}) {
+	config.port = 3420;
+	config.secret = "";
+} else {
+	config = {
+		port: Config.github.port,
+		secret: Config.github.secret,
+	};
+}
+
+const git = exports.github = require('githubhook')(config);
+
+let updates = {};
+
+let sendReport = function (html) {
+	if (Config.github && Config.github.rooms) {
+		for (let curRoom of Config.github.rooms) {
+			let room = Rooms(curRoom);
+			if (room) room.add(`|html|<div class="infobox">${html}</div>`).update();
+		}
+	} else if (Rooms('development')) {
+		Rooms('development').add(`|html|<div class="infobox">${html}</div>`).update();
+	}
+};
+
+git.on('push', (repo, ref, result) => {
+	let url = result.compare;
+	let branch = /[^/]+$/.exec(ref)[0];
+	let messages = [];
+	let message = "";
+	message += `[<font color='FF00FF'>${Chat.escapeHTML(repo)}</font>] `;
+	message += `<font color='909090'>${Chat.escapeHTML(result.pusher.name)}</font> `;
+	message += (result.forced ? '<font color="red">force-pushed</font>' : 'pushed') + " ";
+	message += `<b>${Chat.escapeHTML(result.commits.length)}</b> `;
+	message += `new commit${(result.commits.length === 1 ? '' : 's')} to `;
+	message += `<font color='800080'>${Chat.escapeHTML(branch)}</font>: `;
+	message += `<a href="${Chat.escapeHTML(url)}">View &amp; compare</a>`;
+	messages.push(message);
+	result.commits.forEach(function (commit) {
+		let commitMessage = commit.message;
+		let shortCommit = /.+/.exec(commitMessage)[0];
+		if (commitMessage !== shortCommit) {
+			shortCommit += '&hellip;';
+		}
+		message = "";
+		message += `<font color='FF00FF'>${Chat.escapeHTML(repo)}</font>/`;
+		message += `<font color='800080'>${Chat.escapeHTML(branch)}</font> `;
+		message += `<a href="${Chat.escapeHTML(commit.url)}">`;
+		message += `<font color='606060'>${Chat.escapeHTML(commit.id.substring(0, 6))}</font></a> `;
+		message += `<font color='909090'>${Chat.escapeHTML(commit.author.name)}</font>: ${Chat.escapeHTML(shortCommit)}`;
```

Comment:
>can you explain a bit?

So, say we have template string: `` `Hello, I am <strong>Bob</strong>, but this is ${'<u>'}Sally${'</u>'}`; ``

By doing ``Chat.html`Hello, I am <strong>Bob</strong>, but this is ${'<u>'}Sally${'</u>'}`;`` - this would escape everything surrounded by ${} - so all variables.  This would make it so you don't have to use ``Chat.escapeHTML`` for each instance in the string where HTML could be injected. 

``Chat.html`` in this case would NOT escape the ``<strong>`` tags in my example.

Did that help?